azure/packages/test/scenario-runner/src/AzureClientRunner.ts

```diff
@@ -24,30 +24,48 @@ export interface AzureClientRunnerConfig {
 	userId?: string;
 	userName?: string;
 }
+export type AzureClientRunnerRunConfig = AzureClientRunnerConfig & IRunConfig;
 
 export class AzureClientRunner extends TypedEventEmitter<IRunnerEvents> implements IRunner {
 	private status: RunnnerStatus = "notStarted";
 	constructor(private readonly c: AzureClientRunnerConfig) {
 		super();
 	}
 
-	public async run(config: IRunConfig): Promise<AzureClient | undefined> {
+	public async run(config: IRunConfig): Promise<AzureClient> {
 		this.status = "running";
 
+		try {
+			const ac = await AzureClientRunner.execRun({
+				...config,
+				...this.c,
+			});
+
+			this.status = "success";
+			return ac;
+		} catch {
+			this.status = "error";
+			throw new Error("Failed to create client");
+		}
+	}
+
+	public async runSync(config: IRunConfig): Promise<AzureClient> {
```

Comment:
The only intended change of runSync vs run is for runSync to run in the same process? How does consumer/user reason about this differentiation (when to use one and not the other)?

Reply:
Yes, because the same process allows for passing along complex outputs from previous stages, like an instance of AzureClient. This would be specifically useful in certain scenarios that repro a bug, not necessarily for load testing I would imagine. More for regression testing, but in a way that UTs can't capture well